packages/x-date-pickers/src/TimeClock/Clock.tsx

```diff
@@ -370,7 +379,7 @@ export function Clock(inProps: ClockProps) {
                 type={type}
                 viewValue={viewValue}
                 isInner={isPointerInner}
-                hasSelected={hasSelected}
+                isBetweenTwoClockValues={isPointerBetweenTwoClockValues}
```

Comment:
The `hasSelected` prop name was very unclear to be.
Since it's used in the `ownerState` of `ClockPointer`, I added a better name to it.